### ISSUE TITLE
Could you please include the dll specified in the module manifest?

### ISSUE THREAD

**Issue by KurtDeGreeff** [zacbrown/PowerKrabsEtw#7]

**Comment by zacbrown** [comment]
Hi Kurt - could you elaborate on which file that is? The release should have everything you need: https://github.com/zacbrown/PowerKrabsEtw/releases/tag/1.0.2-alpha

If you're building from scratch, then the correct files should get dropped into the output directory (bin/x64/Release or bin/x64/Debug).

What OS are you trying to run this on?

**Comment by KurtDeGreeff** [comment]
Hi Zac, i'm running latest W10 Creators Update. When I try to build the solution in VS2017, I get following error:
![image](https://user-images.githubusercontent.com/2629743/32797462-226103a4-c972-11e7-8f78-a67d58bac443.png)

I'm also getting a lot namespace errors..
![image](https://user-images.githubusercontent.com/2629743/32797550-7233ead6-c972-11e7-8231-470fc190529c.png)

Wouldn't it be better to include the compiled dll in  the release, so the module can be used without building the solution in VS..

**Comment by KurtDeGreeff** [comment]
I did restore the Nuget packages.. ;-) if I run it now
![image](https://user-images.githubusercontent.com/2629743/32803843-1f587a80-c985-11e7-9f7c-6cbbc0d4250c.png)

Here is the output after Rebuild:

1>------ Rebuild All started: Project: PowerKrabsEtw, Configuration: Debug x64 ------
1>C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\MSBuild\15.0\Bin\Microsoft.Common.CurrentVersion.targets(1964,5): warning MSB3245: Could not resolve this reference. Could not locate the assembly "O365.Security.Native.ETW, Version=1.0.6524.39748, Culture=neutral, PublicKeyToken=80dc624ca478cf0a, processorArchitecture=AMD64". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors.
1>C:\git\PowerKrabsEtw\PowerKrabsEtw\Internal\PropertyParser\IPropertyParser.cs(4,7,4,11): error CS0246: The type or namespace name 'O365' could not be found (are you missing a using directive or an assembly reference?)
1>C:\git\PowerKrabsEtw\PowerKrabsEtw\Internal\PropertyParser\MicrosoftWindowsKernelNetworkParser.cs(4,7,4,11): error CS0246: The type or namespace name 'O365' could not be found (are you missing a using directive or an assembly reference?)
1>C:\git\PowerKrabsEtw\PowerKrabsEtw\Internal\PropertyParser\MicrosoftWindowsPowerShellParser.cs(4,7,4,11): error CS0246: The type or namespace name 'O365' could not be found (are you missing a using directive or an assembly reference?)
1>C:\git\PowerKrabsEtw\PowerKrabsEtw\Internal\ProviderSpecificHandler\IProviderSpecificHandler.cs(4,7,4,11): error CS0246: The type or namespace name 'O365' could not be found (are you missing a using directive or an assembly reference?)
1>C:\git\PowerKrabsEtw\PowerKrabsEtw\Internal\ProviderSpecificHandler\MicrosoftWindowsDNSClientHandler.cs(4,7,4,11): error CS0246: The type or namespace name 'O365' could not be found (are you missing a using directive or an assembly reference?)
1>C:\git\PowerKrabsEtw\PowerKrabsEtw\Internal\PSEtwFilter.cs(4,7,4,11): error CS0246: The type or namespace name 'O365' could not be found (are you missing a using directive or an assembly reference?)
1>C:\git\PowerKrabsEtw\PowerKrabsEtw\Internal\PSEtwUserProvider.cs(4,7,4,11): error CS0246: The type or namespace name 'O365' could not be found (are you missing a using directive or an assembly reference?)
1>C:\git\PowerKrabsEtw\PowerKrabsEtw\Internal\PSEtwUserTrace.cs(4,7,4,11): error CS0246: The type or namespace name 'O365' could not be found (are you missing a using directive or an assembly reference?)
1>C:\git\PowerKrabsEtw\PowerKrabsEtw\NewEtwCallbackFilter.cs(4,7,4,11): error CS0246: The type or namespace name 'O365' could not be found (are you missing a using directive or an assembly reference?)
1>C:\git\PowerKrabsEtw\PowerKrabsEtw\NewEtwUserProvider.cs(1,7,1,11): error CS0246: The type or namespace name 'O365' could not be found (are you missing a using directive or an assembly reference?)
1>C:\git\PowerKrabsEtw\PowerKrabsEtw\Internal\PropertyParser\PropertyExtractor.cs(4,7,4,11): error CS0246: The type or namespace name 'O365' could not be found (are you missing a using directive or an assembly reference?)
1>C:\git\PowerKrabsEtw\PowerKrabsEtw\TraceEtwProcess.cs(4,7,4,11): error CS0246: The type or namespace name 'O365' could not be found (are you missing a using directive or an assembly reference?)
1>C:\git\PowerKrabsEtw\PowerKrabsEtw\TraceEtwProcess.cs(135,34,135,46): error CS0246: The type or namespace name 'IEventRecord' could not be found (are you missing a using directive or an assembly reference?)
1>C:\git\PowerKrabsEtw\PowerKrabsEtw\Internal\ProviderSpecificHandler\IProviderSpecificHandler.cs(10,9,10,29): error CS0246: The type or namespace name 'IEventRecordDelegate' could not be found (are you missing a using directive or an assembly reference?)
1>C:\git\PowerKrabsEtw\PowerKrabsEtw\Internal\ProviderSpecificHandler\MicrosoftWindowsDNSClientHandler.cs(12,16,12,36): error CS0246: The type or namespace name 'IEventRecordDelegate' could not be found (are you missing a using directive or an assembly reference?)
1>C:\git\PowerKrabsEtw\PowerKrabsEtw\Internal\ProviderSpecificHandler\MicrosoftWindowsDNSClientHandler.cs(17,36,17,48): error CS0246: The type or namespace name 'IEventRecord' could not be found (are you missing a using directive or an assembly reference?)
1>C:\git\PowerKrabsEtw\PowerKrabsEtw\Internal\PSEtwFilter.cs(19,41,19,61): error CS0246: The type or namespace name 'IEventRecordDelegate' could not be found (are you missing a using directive or an assembly reference?)
1>C:\git\PowerKrabsEtw\PowerKrabsEtw\Internal\PSEtwFilter.cs(25,30,25,50): error CS0246: The type or namespace name 'IEventRecordDelegate' could not be found (are you missing a using directive or an assembly reference?)
1>C:\git\PowerKrabsEtw\PowerKrabsEtw\Internal\PSEtwFilter.cs(27,18,27,29): error CS0246: The type or namespace name 'EventFilter' could not be found (are you missing a using directive or an assembly reference?)
1>C:\git\PowerKrabsEtw\PowerKrabsEtw\Internal\PSEtwFilter.cs(11,18,11,29): error CS0246: The type or namespace name 'EventFilter' could not be found (are you missing a using directive or an assembly reference?)
1>C:\git\PowerKrabsEtw\PowerKrabsEtw\Internal\PSEtwFilter.cs(12,23,12,43): error CS0246: The type or namespace name 'IEventRecordDelegate' could not be found (are you missing a using directive or an assembly reference?)
1>C:\git\PowerKrabsEtw\PowerKrabsEtw\Internal\PSEtwFilter.cs(14,30,14,41): error CS0246: The type or namespace name 'EventFilter' could not be found (are you missing a using directive or an assembly reference?)
1>C:\git\PowerKrabsEtw\PowerKrabsEtw\Internal\PSEtwUserProvider.cs(49,18,49,26): error CS0246: The type or namespace name 'Provider' could not be found (are you missing a using directive or an assembly reference?)
1>C:\git\PowerKrabsEtw\PowerKrabsEtw\Internal\PSEtwUserProvider.cs(50,30,50,50): error CS0246: The type or namespace name 'IEventRecordDelegate' could not be found (are you missing a using directive or an assembly reference?)
1>C:\git\PowerKrabsEtw\PowerKrabsEtw\Internal\PSEtwUserProvider.cs(51,30,51,41): error CS0246: The type or namespace name 'EventFilter' could not be found (are you missing a using directive or an assembly reference?)
1>C:\git\PowerKrabsEtw\PowerKrabsEtw\Internal\PSEtwUserProvider.cs(53,41,53,61): error CS0246: The type or namespace name 'IEventRecordDelegate' could not be found (are you missing a using directive or an assembly reference?)
1>C:\git\PowerKrabsEtw\PowerKrabsEtw\Internal\PSEtwUserProvider.cs(65,49,65,69): error CS0246: The type or namespace name 'IEventRecordDelegate' could not be found (are you missing a using directive or an assembly reference?)
1>C:\git\PowerKrabsEtw\PowerKrabsEtw\Internal\PSEtwUserProvider.cs(29,36,29,44): error CS0246: The type or namespace name 'Provider' could not be found (are you missing a using directive or an assembly reference?)
1>C:\git\PowerKrabsEtw\PowerKrabsEtw\Internal\PSEtwUserProvider.cs(39,36,39,44): error CS0246: The type or namespace name 'Provider' could not be found (are you missing a using directive or an assembly reference?)
1>C:\git\PowerKrabsEtw\PowerKrabsEtw\Internal\PSEtwUserProvider.cs(15,18,15,26): error CS0246: The type or namespace name 'Provider' could not be found (are you missing a using directive or an assembly reference?)
1>C:\git\PowerKrabsEtw\PowerKrabsEtw\Internal\PSEtwUserProvider.cs(16,23,16,43): error CS0246: The type or namespace name 'IEventRecordDelegate' could not be found (are you missing a using directive or an assembly reference?)
1>C:\git\PowerKrabsEtw\PowerKrabsEtw\Internal\PSEtwUserTrace.cs(106,49,106,61): error CS0246: The type or namespace name 'IEventRecord' could not be found (are you missing a using directive or an assembly reference?)
1>C:\git\PowerKrabsEtw\PowerKrabsEtw\Internal\PropertyParser\IPropertyParser.cs(11,86,11,98): error CS0246: The type or namespace name 'IEventRecord' could not be found (are you missing a using directive or an assembly reference?)
1>C:\git\PowerKrabsEtw\PowerKrabsEtw\Internal\PropertyParser\MicrosoftWindowsKernelNetworkParser.cs(19,93,19,105): error CS0246: The type or namespace name 'IEventRecord' could not be found (are you missing a using directive or an assembly reference?)
1>C:\git\PowerKrabsEtw\PowerKrabsEtw\Internal\PropertyParser\MicrosoftWindowsKernelNetworkParser.cs(38,38,38,50): error CS0246: The type or namespace name 'IEventRecord' could not be found (are you missing a using directive or an assembly reference?)
1>C:\git\PowerKrabsEtw\PowerKrabsEtw\Internal\PropertyParser\MicrosoftWindowsKernelNetworkParser.cs(39,38,39,50): error CS0246: The type or namespace name 'IEventRecord' could not be found (are you missing a using directive or an assembly reference?)
1>C:\git\PowerKrabsEtw\PowerKrabsEtw\Internal\PropertyParser\MicrosoftWindowsPowerShellParser.cs(18,93,18,105): error CS0246: The type or namespace name 'IEventRecord' could not be found (are you missing a using directive or an assembly reference?)
1>C:\git\PowerKrabsEtw\PowerKrabsEtw\Internal\PropertyParser\MicrosoftWindowsPowerShellParser.cs(29,76,29,88): error CS0246: The type or namespace name 'IEventRecord' could not be found (are you missing a using directive or an assembly reference?)
1>C:\git\PowerKrabsEtw\PowerKrabsEtw\Internal\PropertyParser\PropertyExtractor.cs(35,54,35,66): error CS0246: The type or namespace name 'IEventRecord' could not be found (are you missing a using directive or an assembly reference?)
1>C:\git\PowerKrabsEtw\PowerKrabsEtw\Internal\PropertyParser\PropertyExtractor.cs(80,43,80,51): error CS0246: The type or namespace name 'Property' could not be found (are you missing a using directive or an assembly reference?)
1>C:\git\PowerKrabsEtw\PowerKrabsEtw\Internal\PropertyParser\PropertyExtractor.cs(80,58,80,70): error CS0246: The type or namespace name 'IEventRecord' could not be found (are you missing a using directive or an assembly reference?)
1>C:\git\PowerKrabsEtw\PowerKrabsEtw\Internal\PSEtwUserTrace.cs(24,18,24,27): error CS0246: The type or namespace name 'UserTrace' could not be found (are you missing a using directive or an assembly reference?)
========== Rebuild All: 0 succeeded, 1 failed, 0 skipped ==========

**Comment by zacbrown** [comment]
Ah - have you tried pulling? I've updated the actual package being pulled down from nuget.org. That version of the O365.Security.Native.ETW.dll binary (1.0.6524.39748) was for the private build I released in the earliest release. For reference, the version I see in the csproj file is 1.0.6526.18713.

The latest git commit should be https://github.com/zacbrown/PowerKrabsEtw/commit/8e05b40061a394ead91e35d8e30e58260ee52068.

Let me know if that helps.
